From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>, damon@lists.linux.dev, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20251026182216.118200-2-sj@kernel.org>
In-Reply-To: <20251026182216.118200-1-sj@kernel.org>
Date: Sun, 26 Oct 2025 11:22:06 -0700
Subject: [PATCH 1/8] mm/damon/core: fix wrong comment of damon_call() return timing

damon_call() works asynchronously and synchronously for repeat and
non-repeat mode requests, respectively.  The comment about the behavior
is wrong, though.  Fix it.

The wrong comment was introduced together with the repeat mode, by
commit 43df7676e550 ("mm/damon/core: introduce repeat mode
damon_call()").

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20251026182216.118200-2-sj@kernel.org
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index d78f4452e536..06ad359024ad 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1448,7 +1448,7 @@ bool damon_is_running(struct damon_ctx *ctx)
  * Ask DAMON worker thread (kdamond) of @ctx to call a function with an
  * argument data that respectively passed via &damon_call_control->fn and
  * &damon_call_control->data of @control.  If &damon_call_control->repeat of
- * @control is set, further wait until the kdamond finishes handling of the
+ * @control is unset, further wait until the kdamond finishes handling of the
  * request.  Otherwise, return as soon as the request is made.
  *
  * The kdamond executes the function with the argument in the main loop, just
-- 
2.47.3